src/plots/scatterPlot.ts

```diff
@@ -205,4 +222,105 @@ export class Scatter<X, Y> extends XYPlot<X, Y> {
       return x - size / 2 <= p.x && p.x <= x + size / 2 && y - size / 2 <= p.y && p.y <= y + size / 2;
     });
   }
+
+  // labels
+
+  protected _createNodesForDataset(dataset: Dataset): ProxyDrawer {
+    const drawer = super._createNodesForDataset(dataset);
+    const labelArea = this._renderArea.append("g").classed(Scatter._LABEL_AREA_CLASS, true);
+    const context = new Typesettable.SvgContext(labelArea.node() as SVGElement);
+    const measurer = new Typesettable.CacheMeasurer(context);
+    const writer = new Typesettable.Writer(measurer, context);
+    this._labelConfig.set(dataset, { labelArea: labelArea, measurer: measurer, writer: writer });
+    return drawer;
+  }
+
+  protected _removeDatasetNodes(dataset: Dataset) {
+    super._removeDatasetNodes(dataset);
+    const labelConfig = this._labelConfig.get(dataset);
+    if (labelConfig != null) {
+      labelConfig.labelArea.remove();
+      this._labelConfig.delete(dataset);
+    }
+  }
+
+  protected _additionalPaint(time: number) {
+    this.datasets().forEach((dataset) => this._labelConfig.get(dataset).labelArea.selectAll("g").remove());
+    if (this._labelsEnabled) {
+      Utils.Window.setTimeout(() => this._drawLabels(), time);
```

Comment:
This is how labels are done in Bar Plot, from what I can guess it's because it allows labels to be drawn after all of the bars/scatter symbols are done?